### PR TITLE
ENH: Generate Simulation

### PR DESCRIPTION
**Pull request by teddyrendahl**
Create a simulation from a `DataFrame` of collected data

### CODE REVIEW

**Comment by teddyrendahl** [comment]
Thinking out loud, we need a way to modify the distribution during a run. Just aligning to a single data set is boring, we want to be running our scan, optimizing then suddenly swap the jet position (i.e another dataset).

In other words, we need to make the dataframe editable after simulation creation.

File: jet_tracking/sim.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    random_state: np.random.RandomState, optional
+        Seed the simulation
+    """
+    # Create our motor that will serve as the independent variable
+    motor = SynAxis(name=motor_column, precision=motor_precision)
+
+    # Create a function to return a random value from closest motor position
+    motor_positions = dataframe[motor_column].unique()
+    sim_data = dict(iter(dataframe.groupby(motor_column)))
```
Comment (klauer) [inline jet_tracking/sim.py]: Fancy!

**Comment by klauer** [comment]
That's a good point. A bit more flexibility would be useful. When we - fingers crossed - get our data on Monday, we can use it as-is and but also transform it for different scenarios (or jet positions).

File: jet_tracking/sim.py (hunk 1 above)
Comment (teddyrendahl) [reply jet_tracking/sim.py]: 🤓

**Comment by teddyrendahl** [comment]
Changed back to `NameSpace` concept. Should be able to mutate data now and see response in simulation.

**Pull request merged**